connectors/sync_job_runner.py

```diff
@@ -186,6 +193,33 @@ async def execute(self):
             if self.data_provider is not None:
                 await self.data_provider.close()
 
+    async def _update_native_connector_authentication(self):
+        """
+        The connector secrets API endpoint can only be accessed by the Enterprise Search system role,
+        so we need to use a client initialised with the config's username and password to first fetch
+        the API key for native connectors.
+        After that, we can provide the API key to the sync orchestrator to initialise a new client
+        so that an API key can be used for the sync.
+        This function should not be run for connector clients.
+        """
+        es_management_client = ESManagementClient(self.es_config)
+        try:
+            self.sync_job.log_debug(
+                f"Checking secrets storage for API key for index [{self.connector.index_name}]..."
+            )
+            api_key = await es_management_client.get_connector_secret(
+                self.connector.api_key_secret_id
+            )
+            self.sync_job.log_debug(
+                f"API key found in secrets storage for index [{self.connector.index_name}], will use this for authorization."
+            )
+            self.es_config["api_key"] = api_key
```

Comment:
Just to check - I'm not sure if `es_config` here is a copy of original config or not. If it's a copy, then this is fine, if it's a reference to original object, then it'll break if multiple connectors are running at the same time.

Can you verify that running multiple connectors in this scenario won't break?

Reply:
@artem-shelkovnikov the `self.es_config` in `SyncJobRunner` is initially deepcopied from the real config [here](https://github.com/elastic/connectors/blob/b155f2a42e8a9457d8ed463c08dca217d7f2846e/connectors/services/job_execution.py#L83-L89), by the [_override_es_config](https://github.com/elastic/connectors/blob/b155f2a42e8a9457d8ed463c08dca217d7f2846e/connectors/services/base.py#L146C1-L159C25) func.
The `es_config` is initially just the [`elasticsearch` section of the config](https://github.com/elastic/connectors/blob/b155f2a42e8a9457d8ed463c08dca217d7f2846e/connectors/services/base.py#L74). So this config update won't affect multiple connectors.
(For sanity's sake, I ran two connectors together with this setup and they ran without issue).